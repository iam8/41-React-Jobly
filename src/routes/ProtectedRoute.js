import React, {useContext} from "react";
import { Redirect } from "react-router-dom";

import UserContext from "../auth/UserContext";


function ProtectedRoute({children}) {
    const {currentUser} = useContext(UserContext);

    if (!currentUser) return <Redirect to="/" />

    return children;
}


export default ProtectedRoute;
