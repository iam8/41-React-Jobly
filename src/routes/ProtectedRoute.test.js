import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import { DemoUserProvider as UserProvider } from "../testUtils";
import ProtectedRoute from "./ProtectedRoute";


test("Renders without crashing", () => {
    render (
        <MemoryRouter>
            <UserProvider>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>
    );
})


test("Matches snapshot when logged in", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});


test("Matches snapshot when logged out", () => {
    const { asFragment } = render(
        <MemoryRouter>
            <UserProvider currentUser={null}>
                <ProtectedRoute />
            </UserProvider>
        </MemoryRouter>,
    );

    expect(asFragment()).toMatchSnapshot();
});