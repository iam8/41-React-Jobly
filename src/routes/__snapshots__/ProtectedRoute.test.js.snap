// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot when logged in 1`] = `<DocumentFragment />`;

exports[`Matches snapshot when logged out 1`] = `<DocumentFragment />`;
